File: Eldar Dautović/ridewithme/UI/ridewithme_mobile/lib/models/chat_message.dart
// -- chat_message.dart --
import 'package:json_annotation/json_annotation.dart';
import 'package:ridewithme_mobile/models/korisnik.dart';

part 'chat_message.g.dart';

@JsonSerializable()
class ChatMessage {
  int? id;
  String? message;
  Korisnik? sender;
  Korisnik? reciever;
  DateTime? datumKreiranja;

  ChatMessage(
    this.id,
    this.message,
    this.sender,
    this.reciever,
    this.datumKreiranja,
  );

  factory ChatMessage.fromJson(Map<String, dynamic> json) =>
      _$ChatMessageFromJson(json);

  Map<String, dynamic> toJson() => _$ChatMessageToJson(this);
}

File: Eldar Dautović/ridewithme/ridewithme.Model/Models/ChatMessage.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ridewithme.Model.Models
{
    public class ChatMessage
    {
        public int Id { get; set; }
        public string Message { get; set; }

        public Korisnici Sender { get; set; }

        public Korisnici Reciever { get; set; }

        public DateTime DatumKreiranja { get; set; }
    }
}
